salt/transport/base.py

```diff
@@ -54,10 +58,25 @@ def publish_server(opts, **kwargs):
     # Default to ZeroMQ for now
     ttype = "zeromq"
     # determine the ttype
-    if "transport" in opts:
+    if "transport" in kwargs:
+        ttype = kwargs.pop("transport")
+    elif "transport" in opts:
         ttype = opts["transport"]
     elif "transport" in opts.get("pillar", {}).get("master", {}):
         ttype = opts["pillar"]["master"]["transport"]
+
+    if "pub_host" not in kwargs and "pub_path" not in kwargs:
+        kwargs["pub_host"] = opts["interface"]
+    if "pub_port" not in kwargs and "pub_path" not in kwargs:
+        kwargs["pub_port"] = opts.get("publish_port", 4506)
```

Comment:
I know these port values are set in stone, but the old 'C' programmer in me hates magic numbers.
Wondering if we should have these defined global named values, and replace them when come across them.

Reply:
yeah, by now they are pretty much in written in stone, but I'm not against a `salt/defaults/ports.py` file.
We could, however, address that in a separate PR since this one is already quite big.